scipy/signal/_signaltools.py

```diff
@@ -947,9 +984,11 @@ def oaconvolve(in1, in2, mode="full", axes=None):
     # if necessary.
     if not all(curpad == (0, 0) for curpad in pad_size1):
         in1 = np.pad(in1, pad_size1, mode='constant', constant_values=0)
+        in1 = xp.asarray(in1)   # FIXME: xp.pad
```

Comment:
For completeness, `pad(...., mode="constant")` is doable under pytorch: https://github.com/pytorch/pytorch/blob/main/torch/_numpy/_funcs_impl.py#L2045
Not sure about other modes though.
So might be something to keep in mind in case it shows up elsewhere in scipy. Am going to reword the comment and add a link unless there's a strong opinion to the contrary.

Reply:
We have bumped into pad once before, but fortunately just in a test. https://github.com/scipy/scipy/blob/8971d5e9b72931987b7d3c5a25da1a8e7e5485d0/scipy/fft/tests/test_real_transforms.py#L40